tests/serde-de.rs

```diff
@@ -1639,3 +1639,134 @@ mod xml_prolog {
         );
     }
 }
+
+mod nil {
+    use quick_xml::DeError;
+
+    #[derive(Debug, serde::Deserialize, PartialEq)]
+    struct Foo {
+        tag: String,
+    }
+
+    #[derive(Debug, serde::Deserialize, PartialEq)]
+    struct Bar {
+        foo: Option<Foo>,
+    }
+
+    #[derive(Debug, PartialEq, serde::Deserialize)]
+    struct Helper {
+        #[serde(rename = "@attr")]
+        attr: String,
+        #[serde(rename = "$value")]
+        inner: Option<Foo>,
+    }
```

Comment:
Unfortunately, this is incorrect representation for `<foo attr="..." xsi:nil="..."/>`. Currently, you even cannot use `$value` name for annotating fields with non-enum type, the serialization of such fields would fail. So currently I searching a correct way to represent such XML by the Rust type.